modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcTransactionTest.java

```diff
@@ -0,0 +1,307 @@
+            MatcherAssert.assertThat(rollbackEx.getMessage(),
+                    containsString("Transaction cannot be rolled back explicitly in auto-commit mode."));
+
+            conn.setAutoCommit(false);
+            conn.rollback();
+
+            conn.close();
+
+            // Exception when called on closed connection
+            checkConnectionClosed(conn::rollback);
+        }
+    }
+
+    /**
+     * Ensures that {@link Statement#executeUpdate(String)} supports explicit transaction.
+     *
+     * <p>It is expected that in non auto-commit mode, data updates can be rolled back and
+     * are invisible outside of an active transaction until they are committed.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testExecuteUpdate() throws Exception {
+        try (Statement stmt = conn.createStatement()) {
+            checkUpdate(2, 1, (id, ignore) -> stmt.executeUpdate(String.format(SQL_INSERT, id, "name-" + id)));
+        }
+    }
+
+    /**
+     * Ensures that {@link PreparedStatement#executeUpdate()} supports explicit transaction.
+     *
+     * <p>It is expected that in non auto-commit mode, data updates can be rolled back and
+     * are invisible outside of an active transaction until they are committed.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testExecuteUpdatePrepared() throws Exception {
+        try (PreparedStatement pstmt = conn.prepareStatement(SQL_INSERT_PREPARED)) {
+            checkUpdate(2, 1, (id, ignore) -> {
+                pstmt.setInt(1, id);
+                pstmt.setString(2, "name-" + id);
+
+                assertEquals(1, pstmt.executeUpdate());
+            });
+        }
+    }
+
+    /**
+     * Ensures that {@link Statement#executeBatch()} supports explicit transaction.
+     *
+     * <p>It is expected that in non auto-commit mode, data updates can be rolled back and
+     * are invisible outside of an active transaction until they are committed.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testBatch() throws Exception {
+        try (Statement stmt = conn.createStatement()) {
+            checkUpdate(5, 10, (off, cnt) -> {
+                for (int i = off; i < off + cnt; i++) {
+                    stmt.addBatch(String.format(SQL_INSERT, i, "name-" + i));
+                }
+
+                stmt.executeBatch();
+            });
+        }
+    }
+
+    /**
+     * Ensures that {@link PreparedStatement#executeBatch()} supports explicit transaction.
+     *
+     * <p>It is expected that in non auto-commit mode, data updates can be rolled back and
+     * are invisible outside of an active transaction until they are committed.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testBatchPrepared() throws Exception {
+        try (PreparedStatement pstmt = conn.prepareStatement(SQL_INSERT_PREPARED)) {
+            checkUpdate(5, 10, (off, cnt) -> {
```

Comment:
So, you've passed both rowsPerOp and actual op closure as a params to the checkUpdate to propagate them to checkTxResult to pass rowsPerOp as an input to op closure...

Reply:
Reworked a bit :roll_eyes: